src/common/Throttle.cc

```diff
@@ -725,31 +723,99 @@ TokenBucketThrottle::~TokenBucketThrottle()
 
 void TokenBucketThrottle::set_max(uint64_t m) {
   std::lock_guard<Mutex> lock(m_lock);
-  m_throttle.set_max(m);
+  if (0 == m && 0 != m_avg) {
+    m_throttle.set_max(m_avg);
+  } else {
+    m_throttle.set_max(m);
```

Comment:
So set_max(0) means no burst, right? What if I am going to set a max but the max is less than m_avg?

Reply:
1. Yes. 0 is the default value for burst configuration.
2. The max should never less than m_avg. I'll correct it.

Reply:
@yangdongsheng 
I got it !
But I am not sure if I also need to handle the error in `apply_qos_limit` ?
Thanks.

Reply:
No -- I don't think you should prevent the image from opening (since that would also prevent you from fixing it). I think librbd can log to ```lderr``` if it detects invalid configuration settings.

Reply:
@dillaman OK. Thanks.

Reply:
Hi @dillaman @yangdongsheng 
I'am adding the log. But there is a confusing problem.

As we know, the metadata will be applied twice when we set it by `rbd config image set aaa bbb`. The first time is when we are opening the image. And the second time is to apply the new config.

The problem is: 
When we want to correct the value after we set an invalid one, it will still show the error log ( when opening the image, the image will still apply the old config, which contains the invalid value. )

I think this error log may confuse the users. Does that matter?

Reply:
@iridescent-rsy seems like a necessary evil to me. I think if the log message includes the current values for average and burst it would be less confusing.